src/containers/ShareTopicModal.js

```diff
@@ -39,47 +51,187 @@ export default class ShareTopicModal extends React.Component {
 
   render () {
     const { onCancel, tagName, community, dispatch, pending,
-      tagInvitationEditor: { recipients, error, success } } = this.props
+      tagInvitationEditor: { recipients, recipient, error, success } } = this.props
+
+    const { copied } = this.state
+
     const joinUrl = get(community, 'beta_access_code')
       ? communityTagJoinUrl(community, tagName)
       : null
 
     const setError = text => dispatch(updateTagInvitationEditor('error', text))
 
-    const update = (field) => event =>
-      dispatch(updateTagInvitationEditor(field, event.target.value))
+    const update = (field) => value =>
```

Comment:
if you wrap this in `curry` then you can do away with the `update(...)(...)` calls later